minpy/nn/solver.py

```diff
@@ -4,35 +4,39 @@
 from __future__ import print_function
 import time
 
+import minpy
+
 from minpy.nn import optim, init
 from minpy import core
 import minpy.numpy as np
 
+from minpy.visualize.writer import LegacySummaryWriter as SummaryWriter
+import minpy.visualize.summaryOps as summaryOps
 
 class Solver(object):
     """
     A Solver encapsulates all the logic necessary for training classification
     models. The Solver performs stochastic gradient descent using different
     update rules defined in optim.py.
-
+    
```

Comment:
That's weird. Just have a try in your forked repo:
```
 python ./dmlc-core/scripts/lint.py minpy python minpy/visualize/summaryOps.py                                    ──(Wed,Nov30)─┘
No config file found, using default configuration

************* Module summaryOps
 minpy/visualize/summaryOps.py:14: warning (W0311, bad-indentation, ) Bad indentation. Found 6 spaces, expected 8
 minpy/visualize/summaryOps.py:16: warning (W0311, bad-indentation, ) Bad indentation. Found 6 spaces, expected 8
 minpy/visualize/summaryOps.py:39: warning (W0311, bad-indentation, ) Bad indentation. Found 6 spaces, expected 8
 minpy/visualize/summaryOps.py:40: warning (W0311, bad-indentation, ) Bad indentation. Found 6 spaces, expected 8
 minpy/visualize/summaryOps.py:43: convention (C0330, bad-continuation, ) Wrong continued indentation (add 1 space).
                    if counts[i]]
                    ^|
 minpy/visualize/summaryOps.py:56: warning (W0311, bad-indentation, ) Bad indentation. Found 2 spaces, expected 4
 minpy/visualize/summaryOps.py:57: warning (W0311, bad-indentation, ) Bad indentation. Found 2 spaces, expected 4
 minpy/visualize/summaryOps.py:58: warning (W0311, bad-indentation, ) Bad indentation. Found 2 spaces, expected 4
 minpy/visualize/summaryOps.py:59: warning (W0311, bad-indentation, ) Bad indentation. Found 2 spaces, expected 4
 minpy/visualize/summaryOps.py:60: warning (W0311, bad-indentation, ) Bad indentation. Found 4 spaces, expected 8
 minpy/visualize/summaryOps.py:61: warning (W0311, bad-indentation, ) Bad indentation. Found 4 spaces, expected 8
 minpy/visualize/summaryOps.py:62: warning (W0311, bad-indentation, ) Bad indentation. Found 4 spaces, expected 8
 minpy/visualize/summaryOps.py:63: warning (W0311, bad-indentation, ) Bad indentation. Found 4 spaces, expected 8
 minpy/visualize/summaryOps.py:64: warning (W0311, bad-indentation, ) Bad indentation. Found 4 spaces, expected 8
 minpy/visualize/summaryOps.py:65: warning (W0311, bad-indentation, ) Bad indentation. Found 4 spaces, expected 8
 minpy/visualize/summaryOps.py:66: warning (W0311, bad-indentation, ) Bad indentation. Found 4 spaces, expected 8
 minpy/visualize/summaryOps.py:1: convention (C0103, invalid-name, ) Invalid module name "summaryOps"
 minpy/visualize/summaryOps.py:1: convention (C0111, missing-docstring, ) Missing module docstring
 minpy/visualize/summaryOps.py:7: error (E0401, import-error, ) Unable to import 'minpy'
 minpy/visualize/summaryOps.py:10: error (E0401, import-error, ) Unable to import 'minpy.visualization.summary_pb2'
 minpy/visualize/summaryOps.py:12: convention (C0103, invalid-name, scalarSummary) Invalid function name "scalarSummary"
 minpy/visualize/summaryOps.py:12: convention (C0103, invalid-name, scalarSummary) Invalid argument name "TAG"
 minpy/visualize/summaryOps.py:12: convention (C0111, missing-docstring, scalarSummary) Missing function docstring
 minpy/visualize/summaryOps.py:13: convention (C0121, singleton-comparison, scalarSummary) Comparison to True should be just 'expr' or 'expr is True'
 minpy/visualize/summaryOps.py:15: convention (C0121, singleton-comparison, scalarSummary) Comparison to True should be just 'expr' or 'expr is True'
 minpy/visualize/summaryOps.py:19: convention (C0103, invalid-name, histogramSummary) Invalid function name "histogramSummary"
 minpy/visualize/summaryOps.py:19: convention (C0103, invalid-name, histogramSummary) Invalid argument name "TAG"
 minpy/visualize/summaryOps.py:19: convention (C0111, missing-docstring, histogramSummary) Missing function docstring
 minpy/visualize/summaryOps.py:23: convention (C0103, invalid-name, _MakeHistogramBuckets) Invalid function name "_MakeHistogramBuckets"
 minpy/visualize/summaryOps.py:24: convention (C0103, invalid-name, _MakeHistogramBuckets) Invalid variable name "v"
 minpy/visualize/summaryOps.py:30: convention (C0103, invalid-name, _MakeHistogramBuckets) Invalid variable name "v"
 minpy/visualize/summaryOps.py:34: convention (C0103, invalid-name, _MakeHistogram) Invalid function name "_MakeHistogram"
 minpy/visualize/summaryOps.py:38: convention (C0103, invalid-name, _MakeHistogram) Invalid variable name "v"
 minpy/visualize/summaryOps.py:42: error (E0602, undefined-variable, _MakeHistogram) Undefined variable 'xrange'
 minpy/visualize/summaryOps.py:55: convention (C0103, invalid-name, WriteHistogramSeries) Invalid function name "WriteHistogramSeries"
 minpy/visualize/summaryOps.py:55: convention (C0103, invalid-name, WriteHistogramSeries) Invalid argument name "n"
 minpy/visualize/summaryOps.py:58: error (E0602, undefined-variable, WriteHistogramSeries) Undefined variable '_start_time'
 minpy/visualize/summaryOps.py:60: error (E0602, undefined-variable, WriteHistogramSeries) Undefined variable 'xrange'
 minpy/visualize/summaryOps.py:62: error (E0602, undefined-variable, WriteHistogramSeries) Undefined variable 'tf'
 minpy/visualize/summaryOps.py:62: error (E0602, undefined-variable, WriteHistogramSeries) Undefined variable 'tf'
 minpy/visualize/summaryOps.py:63: error (E0602, undefined-variable, WriteHistogramSeries) Undefined variable 'tf'
 minpy/visualize/summaryOps.py:6: convention (C0411, wrong-import-order, ) standard import "import bisect" comes before "import numpy"
 minpy/visualize/summaryOps.py:8: convention (C0411, wrong-import-order, ) standard import "import random" comes before "import numpy"
```

You should try to fix them. You could leave the `solver.py` file right now until @sneakerkg finished the lint fix on this file.